### ISSUE TITLE
Add onBeforeEditorShow and onBeforeEditorHide callbacks to Column on DataTable with cell editing

### ISSUE THREAD

**Issue by KoushikVanama** [primefaces/primereact#589]
**I'm submitting a ...**
```
[X] bug report
[ ] feature request
[ ] support request
```
**Example:**
In my code, I have a usecase like this
```
<DataTable value={this.state.rows} scrollable="true" onRowClick={rowClicked} editable={true}>
              <Column field="dayPart" header="DAYPART" />
              <Column field="start_time" header="START TIME" editor={this.sliderTemplate} />
              <Column field="name" header="NAME" />
</DataTable>
```
and my sliderTemplate() returns a component and looks like this
```
sliderTemplate = (props) => {    
    return(
      <div>
        <TimeSlider format={12} tableInfo={props} onValueSelectSlider={this.handleSelectedValueTS} />
      </div>
    )
}
```
**Screenshot:**
![image](https://user-images.githubusercontent.com/24358668/46026487-1706fb00-c109-11e8-82fa-cd5d953da9b7.png)

On clicking start_time column field the existing value is disappearing and component is showing in place of it. But how can I make sure that the existing value shows as is and the TimeSlider component shows below to that field.
I think this makes sense. Is it achievable?

**Please tell us about your environment:**

React version: ^16.4.1

PrimeReact version: ^2.0.0-beta.5

Browsers: Chrome 69.0.3497.100v

Language: JSX

**Comment by deepak-ww** [comment]
@KoushikVanama I am also facing the same issue. If you got a solution to this then can you please share it. Thanks.

**Comment by mertsincan** [comment]
Hi,
Added two callbacks to control editing phase;
onBeforeEditorHide?(e: {originalEvent: Event, columnProps: any}) -> Callback to invoke before the cell editor is shown.
onBeforeEditorShow?(e: {originalEvent: Event, columnProps: any}) -> Callback to invoke before the cell editor is hidden.
```
onBeforeEditorHide() {
    //condition
   
    return; // prevent cell closing
}
```

Also, I made some changes related to editing cells on 6.1.0.

Best Regards,
